src/consensus/tx_verify.h

```diff
@@ -24,7 +26,7 @@ namespace Consensus {
  * This does not modify the UTXO set. This does not check scripts and sigs.
  * Preconditions: tx.IsCoinBase() is false.
  */
-bool CheckTxInputs(const CTransaction& tx, CValidationState& state, const CCoinsViewCache& inputs, int nSpendHeight);
+bool CheckTxInputs(const CTransaction& tx, CValidationState& state, const CCoinsViewCache& inputs, int nSpendHeight, CAmount& nFees);
```

Comment:
@jtimon You can 'return` the fee in a tuple, or as a pass-by-reference value. So for example `CheckTxInputs` would have `CAmount& return_fee` field, and `return_fee` would just be *assigned* that transaction's fee. The caller can then do the `accumulated_fee += return_fee` logic. This involves no duplication, and still makes `CheckTxInputs` purely operate on data for a single transaction.